aptos-move/vm-genesis/src/lib.rs

```diff
@@ -317,23 +334,69 @@ fn create_and_initialize_validators(
     );
 }
 
-/// Publish the standard library.
-fn publish_stdlib(session: &mut SessionExt<impl MoveResolver>, stdlib: Modules) {
-    let dep_graph = stdlib.compute_dependency_graph();
+/// Publish all modules that should be available after genesis.
+fn publish_stdlib(session: &mut SessionExt<impl MoveResolver>, stdlib: Vec<CompiledModule>) {
+    let map = construct_module_map(stdlib);
+    let root_address = AccountAddress::from_hex_literal("0x1").unwrap();
+    let token_address = AccountAddress::from_hex_literal("0x2").unwrap();
+
+    let framework_modules = map.get(&root_address).unwrap();
+    let token_modules = map.get(&token_address).unwrap();
+
+    // publish core-framework
+    publish_module_bundle(session, Modules::new(framework_modules));
+    // publish non-core-framework modules
+    publish_token_modules(session, token_modules.clone());
+}
+
+/// publish modules that are not core-framework. assuming core-framework published
+/// the modules has to be sorted by topological order PropertyMap -> TokenV1 -> TokenCoinSwap
+fn publish_token_modules(
```

Comment:
How about we move the token modules into a separate dir so we don't need to maintain this hardcoded list of modules to deploy to 0x2?

Reply:
This is unfortunately the limitation of our publish module functions. 

To publish module bundle, it requires all modules to be from same address. So I have to publish module individually. To publish module separately, I have to handle the topological order myself. The current bundle topological sort fails as the same reason above.

Reply:
Really, conceptually token is not part of the stdlib and shouldn't be considered part of genesis. Once we have the new deployment model, we need to revisit this.